common/persistence/execution_manager.go

```diff
@@ -402,46 +408,157 @@ func (m *executionManagerImpl) SerializeWorkflowMutation(
-		ChildExecutionInfos: input.ChildExecutionInfos,
-		RequestCancelInfos:  input.RequestCancelInfos,
-		SignalInfos:         input.SignalInfos,
-		SignalRequestedIDs:  input.SignalRequestedIDs,
+func (m *executionManagerImpl) SerializeWorkflowSnapshot(
+	input *WorkflowSnapshot,
+) (*InternalWorkflowSnapshot, error) {
 
-		TransferTasks:    input.TransferTasks,
-		ReplicationTasks: input.ReplicationTasks,
-		TimerTasks:       input.TimerTasks,
-		VisibilityTasks:  input.VisibilityTasks,
+	var err error
+	result := &InternalWorkflowSnapshot{
+		NamespaceID: input.ExecutionInfo.GetNamespaceId(),
+		WorkflowID:  input.ExecutionInfo.GetWorkflowId(),
+		RunID:       input.ExecutionState.GetRunId(),
+
+		ActivityInfos:       make(map[int64]*commonpb.DataBlob),
+		TimerInfos:          make(map[string]*commonpb.DataBlob),
+		ChildExecutionInfos: make(map[int64]*commonpb.DataBlob),
+		RequestCancelInfos:  make(map[int64]*commonpb.DataBlob),
+		SignalInfos:         make(map[int64]*commonpb.DataBlob),
+
+		ExecutionState:     input.ExecutionState,
+		SignalRequestedIDs: input.SignalRequestedIDs,
+		TransferTasks:      input.TransferTasks,
+		ReplicationTasks:   input.ReplicationTasks,
+		TimerTasks:         input.TimerTasks,
+		VisibilityTasks:    input.VisibilityTasks,
 
 		Condition:       input.Condition,
 		DBRecordVersion: input.DBRecordVersion,
-		Checksum:        input.Checksum,
-	}, nil
+		NextEventID:     input.NextEventID,
+
+		StartVersion: input.ExecutionInfo.StartVersion,
+	}
+
+	result.ExecutionInfo, err = m.serializer.WorkflowExecutionInfoToBlob(input.ExecutionInfo, enumspb.ENCODING_TYPE_PROTO3)
+	if err != nil {
+		return nil, err
+	}
+	result.ExecutionStateBlob, err = m.serializer.WorkflowExecutionStateToBlob(input.ExecutionState, enumspb.ENCODING_TYPE_PROTO3)
+	if err != nil {
+		return nil, err
+	}
+	result.LastWriteVersion, err = getLastWriteVersion(input.ExecutionInfo.VersionHistories)
+	if err != nil {
+		return nil, err
+	}
+
+	for key, info := range input.ActivityInfos {
+		blob, err := m.serializer.ActivityInfoToBlob(info, enumspb.ENCODING_TYPE_PROTO3)
+		if err != nil {
+			return nil, err
+		}
+		result.ActivityInfos[key] = blob
+	}
+	for key, info := range input.TimerInfos {
+		blob, err := m.serializer.TimerInfoToBlob(info, enumspb.ENCODING_TYPE_PROTO3)
+		if err != nil {
+			return nil, err
+		}
+		result.TimerInfos[key] = blob
+	}
+	for key, info := range input.ChildExecutionInfos {
+		blob, err := m.serializer.ChildExecutionInfoToBlob(info, enumspb.ENCODING_TYPE_PROTO3)
+		if err != nil {
+			return nil, err
+		}
+		result.ChildExecutionInfos[key] = blob
+	}
+	for key, info := range input.RequestCancelInfos {
+		blob, err := m.serializer.RequestCancelInfoToBlob(info, enumspb.ENCODING_TYPE_PROTO3)
+		if err != nil {
+			return nil, err
+		}
+		result.RequestCancelInfos[key] = blob
+	}
+	for key, info := range input.SignalInfos {
+		blob, err := m.serializer.SignalInfoToBlob(info, enumspb.ENCODING_TYPE_PROTO3)
```

Comment:
can we set the default serialization format within execution manager instead of hard code here?